eppo_core/src/ufc/eval_details.rs

```diff
@@ -0,0 +1,102 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use crate::{AttributeValue, Attributes};
+
+use super::{AssignmentValue, Condition, FlagEvaluationError, Shard, Value};
+
+/// Details about feature flag evaluation.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvalFlagDetails {
+    pub flag_key: String,
+    pub subject_key: String,
+    pub subject_attributes: Attributes,
+    /// Timestamp when the flag was evaluated.
+    pub timestamp: DateTime<Utc>,
+    /// Details of configuration used for evaluation. None if configuration hasn't been fetched yet.
+    pub configuration_details: Option<ConfigurationDetails>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub result: Option<AssignmentValue>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<FlagEvaluationError>,
+    /// Key of the selected variation.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variation_key: Option<String>,
+    /// Value of the selected variation. Could be `None` if no variation is selected, or selected
+    /// value is absent in configuration (configuration error).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub variation_value: Option<Value>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub allocations: Vec<EvalAllocationDetails>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ConfigurationDetails {
+    /// Timestamp when configuration was fetched by the SDK.
+    pub fetched_at: DateTime<Utc>,
+    /// Timestamp when configuration was published by the server.
+    pub published_at: DateTime<Utc>,
+    /// Environment the configuration belongs to.
+    pub environment_name: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EvalAllocationDetails {
```

Comment:
This is missing `orderPosition` which seems redundant given allocations are in-order in `EvalFlagDetails.allocations`. I can easily add it though, and it might make sense given it's 1-based and is displayed on UI.

Reply:
Yeah, might as well since it's also talked about in the documentation. It might also make it easier to identify visually rather than having to count / keep track of which allocation is in which position. It's a small thing, but also easy to add.